src/sage/libs/ntl/ntl_ZZ_pX.pyx

```diff
@@ -489,23 +490,78 @@ cdef class ntl_ZZ_pX():
         sig_off()
         return r
 
-    def __pow__(ntl_ZZ_pX self, long n, ignored):
+    def __pow__(self, n, modulus):
         """
         Return the n-th nonnegative power of self.
 
+        If ``modulus`` is not ``None``, the exponentiation is performed
+        modulo the polynomial ``modulus``.
+
         EXAMPLES::
 
             sage: c = ntl.ZZ_pContext(20)
             sage: g = ntl.ZZ_pX([-1,0,1],c)
             sage: g**10
             [1 0 10 0 5 0 0 0 10 0 8 0 10 0 0 0 5 0 10 0 1]
+
+            sage: x = ntl.ZZ_pX([0,1],c)
+            sage: x**10
+            [0 0 0 0 0 0 0 0 0 0 1]
+
+        Modular exponentiation::
+
+            sage: c = ntl.ZZ_pContext(20)
+            sage: f = ntl.ZZ_pX([1,0,1],c)
+            sage: m = ntl.ZZ_pX([1,0,0,0,0,1],c)
+            sage: pow(f, 123**45, m)
+            [1 19 3 0 3]
+
+        Modular exponentiation of ``x``::
+
+            sage: f = ntl.ZZ_pX([0,1],c)
+            sage: f.is_x()
+            True
+            sage: m = ntl.ZZ_pX([1,1,0,0,0,1],c)
+            sage: pow(f, 123**45, m)
+            [15 5 5 11]
+        """
+        if modulus is None:
+            return self._pow(n)
+        else:
+            return self._powmod(Integer(n), modulus)
+
+    cpdef ntl_ZZ_pX _pow(ntl_ZZ_pX self, n):
+        """
+        Compute the n-th power of self.
```

Comment:
```suggestion
    cdef ntl_ZZ_pX _pow(ntl_ZZ_pX self, long n):
        """
        Compute the ``n``-th power of `self``.
```
If this is not a `cdef` (but a `cpdef`) then it definitely needs doctests. I doubt this will be called from outside of Cython.

You're also doing a cast in the code, so I would make it explicit here.